plugins/woocommerce/includes/wc-order-functions.php

```diff
@@ -899,6 +912,12 @@ function wc_update_total_sales_counts( $order_id ) {
 add_action( 'woocommerce_order_status_completed', 'wc_update_total_sales_counts' );
 add_action( 'woocommerce_order_status_processing', 'wc_update_total_sales_counts' );
 add_action( 'woocommerce_order_status_on-hold', 'wc_update_total_sales_counts' );
+add_action( 'woocommerce_order_status_completed_to_cancelled', 'wc_update_total_sales_counts' );
+add_action( 'woocommerce_order_status_completed_to_refunded', 'wc_update_total_sales_counts' );
+add_action( 'woocommerce_order_status_processing_to_cancelled', 'wc_update_total_sales_counts' );
+add_action( 'woocommerce_order_status_processing_to_refunded', 'wc_update_total_sales_counts' );
+add_action( 'woocommerce_order_status_on-hold_to_cancelled', 'wc_update_total_sales_counts' );
+add_action( 'woocommerce_order_status_on-hold_to_refunded', 'wc_update_total_sales_counts' );
```

Comment:
A problem we have (and though I don't think it is new as of this PR, these added lines will exacerbate things) is that calling `wp_update_total_sales_count()` each time the order status changes leads to the `total_sales` value changing inappropriately. Example:

| Event | `total_sales` |
| ----- | ------------- |
| Create a new product with 1,000 units of inventory | 0 |
| Place an order for 10 units | 10 |
| Edit the order, change the status to completed | 20 |
| Edit again, change the status to processing | 30 |
| Edit again, change the status to completed | 40 |
| *Process repeats...* | *...* |

That isn't ideal, and we should probably also address it (not necessarily from this PR, but I think it would make sense to try and make these fixes within the same release).

✍🏼 If you have WP CLI available, you can run the following command to check the value of `total_sales` for a specific product (change 1234 to your product ID):

```sh
wp db query "SELECT meta_value FROM $(wp db prefix)postmeta WHERE post_id = 1234 AND meta_key = 'total_sales'"
```

Reply:
@barryhughes Thank you for your feedback!

I tested this scenario, I can see its not increased, because we check (recorded_sales) so it will be increased once until its decreased (if cancelled)

	`if ( $recorded_sales && ! $reflected_order ) {
		return;
	}`

so this is what happen now :

| Event  | `total_sales` |
| ------------- | ------------- |
| Place an order for 10 units  | 10  |
| Edit the order, change the status to completed  | 10  |
| Edit again, change the status to processing  | 10  |
| Edit the order, change the status to completed  | 10  |
| Edit the order, change the status to cancelled  | **0**  |
| Edit the order, change the status to processing  | 10  |
| Edit the order, change the status to completed  | 10  |
| Process repeats...  | ...  |

Reply:
I see what the problem is now, and it's unrelated to your change: when HPOS is enabled along with sync, you should be able to see the problem I described.

It does already happen even without this change, though, so I don't think that needs to block this work 👍🏼